File: CPP/Program/Cpu.cpp
#include "Komponen.cpp"
#include "Connectable.cpp"

using namespace std;

class Cpu : public Komponen, public Connectable {
    private:
        int jumlahCore;
        float kecepatanGHz;
        string socketType;
    public:
        Cpu(){}

        Cpu(string merk, string nama, int jumlahCore, float kecepatanGHz, string socketType) 
            : Komponen(merk, nama) {
            this->jumlahCore = jumlahCore;
            this->kecepatanGHz = kecepatanGHz;
            this->socketType = socketType;
        }

        // setter
        void setJumlahCore(int jumlahCore){ this->jumlahCore = jumlahCore; }
        void setKecepatanGHz(float kecepatanGHz){ this->kecepatanGHz = kecepatanGHz; }
        void setSocketType(string socketType){ this->socketType = socketType; }

        // getter
        int getJumlahCore(){ return jumlahCore; }
        float getKecepatanGHz(){ return kecepatanGHz; }
        string getSocketType(){ return socketType; }

        // Implementasi dari Connectable
        string getConnectionType() override {
            return "Socket Type: " + socketType;
        }

        void displayInfo() override {   // Override method
            cout  << getNama() << " (" << jumlahCore << " cores, " << kecepatanGHz << " GHz, Socket Type: " << socketType << ")" << endl;
        }

        ~Cpu(){}
};

File: Python/program/Cpu.py
from Komponen import Komponen
from Connectable import Connectable

class Cpu(Komponen, Connectable):
    def __init__(self, merk, nama, jumlahCore, kecepatanGHz, socketType):
        super().__init__(merk, nama)
        self.jumlahCore = jumlahCore
        self.kecepatanGHz = kecepatanGHz
        self.socketType = socketType

    # Setter
    def setJumlahCore(self, jumlahCore):
        self.jumlahCore = jumlahCore

    def setKecepatanGHZ(self, kecepatanGHz):
        self.kecepatanGHz = kecepatanGHz

    def setSocketType(self, socketType):
        self.socketType = socketType

    # Getter
    def getJumlahCore(self):
        return self.jumlahCore

    def getKecepatanGHZ(self):
        return self.kecepatanGHz

    def getSocketType(self):
        return self.socketType

    # Implementasi dari Connectable
    def getConnectionType(self):
        return f"{self.socketType}"

    def displayInfo(self):
        print(f"{self.getNama()} ({self.jumlahCore} cores, {self.kecepatanGHz} GHz, Socket Type: {self.socketType})")
